### ISSUE TITLE
[Spree upgrade] Impossible to update stock info (on hand and on demand)

### ISSUE THREAD

**Issue by RachL** [openfoodfoundation/openfoodnetwork#3573]
## Description
I can't see the field in order to update a stock info in a variant. When creating a product everything is ok.

## Steps to Reproduce
1. Go to https://staging.openfoodfrance.org/admin/products
2. Edit a variant. E.g. https://staging.openfoodfrance.org/admin/products/apple-cider-1-2/variants/12/edit
3. You won't see the on hand stock field neither the on demand checkbox. 

## Animated Gif/Screenshot
![image](https://user-images.githubusercontent.com/6525576/53738486-9a363500-3e8f-11e9-9ec2-9f806fc984bb.png)

## Severity
bug-s1: a critical feature is broken: checkout, payments, signup, login

## Your Environment

* Version used: ??
* Browser name and version: Firefox 64
* Operating System and version (desktop or mobile): Ubuntu 18.10

**Comment by kreopelle** [comment]
@RachL  - Is this issue still open?

**Comment by RachL** [comment]
@kreopelle Hello Kayla and welcome to the community :-) yes this issue is still open. However if you are looking for something to start with, I suggest you take a look at the issues labeled "good first issue" : https://github.com/openfoodfoundation/openfoodnetwork/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

This one is part of an upgrade of our main dependencies and may required to know a bit more our platform before working on it.

If you need anything, do not hesitate to ask around here or join [our slack](https://openfoodnetwork.slack.com/messages/C02G54U7H/(url)) channel

**Comment by luisramos0** [comment]
there's a new page called stock management for this! I just discovered it...

![screenshot 2019-03-07 at 22 49 09](https://user-images.githubusercontent.com/1640378/53995247-6fb4d800-412c-11e9-81a0-f78094df7676.png)

I just need to find out how to set the right permissions so we can see the menu option enabled.

**Comment by luisramos0** [comment]
This spree v2 page has two problems:
- it handles stock locations but because there's only one stock location in ofn v2, it's just a select box with one option always.
- also, on_demand is backorderable here. backorderable is like on_demand BUT you can manage stock even if backorderable is activated... on_demand in other parts of ofn v2 is used to disable on_hand.
I'lll see what's the best/easiest way out of these problems...

**Comment by kreopelle** [comment]
Thanks @RachL ! I'll look for another :)

**Comment by luisramos0** [comment]
hey @RachL the stock management menu entry is available, no permissions changes are needed.
you can have a look in AU staging now: https://staging.openfoodnetwork.org.au/admin/products/potatoes-1/stock

This page actually works well! You can change stock with deltas only here, you can add negative values to decrease for example. The stock location is always default and it's not a functional problem, **just** UX...

The backorderable problem is also just a naming, because this field is actually on_demand.

The delete button works and resets stock to zero.
You can play with this page and see how it interacts with the bulk product edit page fields.

So, the easiest way out of this is to rename "Backorderable" to "On demand" and leave everything as is...

**Comment by RachL** [comment]
@luisramos0 ok for this, however while playing with it I saw 3 weird things:

1. In the stock management menu, the "Variant" dropdown is always empty no matter the number of variants the product has
![image](https://user-images.githubusercontent.com/6525576/54375334-22cb8700-4681-11e9-9c9f-01b712c65c85.png)

So I never managed to "add stock" this way. Also I'm a bit confused: isn't editing the figure below enough? I would suggest we hide this upper part, what do you think?

2. I noticed that if I delete a line in the stock management menu, the stock of the variant does indeed change to 0 but afterwards I can't edit it anymore through the main page. I get a 500 error:

![image](https://user-images.githubusercontent.com/6525576/54375670-c9178c80-4681-11e9-9ced-0de4383d09a8.png)

And I can't do it in the edit / stock management menu anymore as I deleted the line.
We can either remove the delete option or fix this.

3. When you add stock info and then change to on_demand, the stock info stays. We should have the same display as we have on main page (=> "on demand writen in on_hand column). Don't you think?

![image](https://user-images.githubusercontent.com/6525576/54376023-6c68a180-4682-11e9-8df1-265a5781351d.png)

Also : can you describe me the use of the "stock location" field?

**Comment by luisramos0** [comment]
ah, nice!

1. the dropdown shows the variant sku, if you set the sku in the variants, this dropdown will make sense... maybe we can change this to show a combination of unit_descritpion and sku... or maybe put the variant ID so this is never empty.

2. I tested something similar but didnt try to update it in the bulk page in the end. I can replicate. I'll have a look at this.

3. spree backorderable let's you change on_hand. but I agree, we should change this to on_demand and hide on_hand field when on_demand is set.

4. re stock location, currently nothing can be done with it, there's only one default stock location in the system.

Let's agree on what we do with point 1 and 3.

**Comment by RachL** [comment]
@luisramos0 

On 1. Can we just used unit_description? I believe this field has always data, whereas sku.

On 3. It's too complex to display the text "On demand" in the on hand field if on demand is activated (like on main page)? If yes, then I agree to hide the field.

On more thing Luis, I just added a new enterprise on staging and created a product, but I don't have the stock management menu. What do I need to do to activate it?

![image](https://user-images.githubusercontent.com/6525576/54377580-b010da80-4685-11e9-8046-a9d35105ee5d.png)
